packages/proxy/lib/http/response-middleware.ts

```diff
@@ -306,7 +308,35 @@ const OmitProblematicHeaders: ResponseMiddleware = function () {
     'connection',
   ])
 
-  this.res.set(headers)
+  this.debug('The headers are %o', headers)
+
+  // Filter for invalid headers
+  const filteredHeaders = Object.fromEntries(
+    Object.entries(headers).filter(([key, value]) => {
+      try {
+        validateHeaderName(key)
+        if (Array.isArray(value)) {
+          value.forEach((v) => validateHeaderValue(key, v))
+        } else {
```

Comment:
```suggestion
        } else if (value) {
```

`value` is possibly undefined; this probably needs an additional `else` to handle this case.

Reply:
I though that giving an undefined value into `validateHeaderValue` would make it raise the error with code `ERR_INVALID_HEADER_VALUE`, since this is the expected error in the Node docs. After some further research i found out that typescript excepts the value type to be string, therefore creating an error. 

I will be raising a warning when the value is undefined, but i realised there's also an invalid character error. I think adding it to the `PROXY_ENCOUNTERED_INVALID_HEADER_VALUE` error fits perfectly to the category of the error, since there's the original error attached, letting the user know if its either an undefined value or an invalid character.

The outcome of my changes are the following:
`
// Filter for invalid headers
  const filteredHeaders = Object.fromEntries(
    Object.entries(headers).filter(([key, value]) => {
      try {
        validateHeaderName(key)
        if (Array.isArray(value)) {
          value.forEach((v) => validateHeaderValue(key, v))
        } else if (value !== undefined) {
          validateHeaderValue(key, value)
        } else {
          error.warning('PROXY_ENCOUNTERED_INVALID_HEADER_VALUE', { [key]: value }, this.req.method, this.req.originalUrl, new TypeError('Header value is undefined while expecting string'));
          return false;
        }
        return true
      } catch (err) {
        if (err.code === 'ERR_INVALID_HTTP_TOKEN') {
          error.warning('PROXY_ENCOUNTERED_INVALID_HEADER_NAME', { [key]: value }, this.req.method, this.req.originalUrl, err)
        } else if (err.code === 'ERR_INVALID_CHAR') {
          error.warning('PROXY_ENCOUNTERED_INVALID_HEADER_VALUE', { [key]: value }, this.req.method, this.req.originalUrl, err)
        } else {
            // rethrow any other errors
            throw err;
        }
        return false
      }
    }),
  )
`
  
 I have also ran the system-tests by using the `yarn test` command in the root of cypress and had no errors.

Reply:
That looks like a great solution! Feel free to update this PR with that change, so that CI will pass the `check-ts` step. You'll also need to update the changelog at this point.

Thank you!